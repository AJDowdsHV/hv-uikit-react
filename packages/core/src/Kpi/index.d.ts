export { default } from "./Kpi";
export * from "./Kpi";
