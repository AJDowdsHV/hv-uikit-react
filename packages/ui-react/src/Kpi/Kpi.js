/**
 * Copyright (c) 2018 Hitachi Vantara Corporation.
 *
 *  The copyright to the computer software herein is the property of
 *  Hitachi Vantara Corporation. The software may be used and/or copied only
 *  with the written permission of Hitachi Vantara Corporation or in accordance
 *  with the terms and conditions stipulated in the agreement/contract
 *  under which the software has been supplied.
 */

import React from "react";
import PropTypes from "prop-types";
import classNames from "classnames";
import Typography from "@material-ui/core/Typography";

const HvKpi = props => {
  const {
    classes,
    kpiTextConfiguration
  } = props;

  return (
    <div className={classes.kpiContainer}>
      <Typography variant="subtitle2">{kpiTextConfiguration.title}</Typography>
      <div className={classes.indicatorsContainer}>
        <Typography className={classNames(classes.indicator, classes.spacingToTheRight)} variant="h1">{kpiTextConfiguration.indicator}</Typography>
        <Typography className={classes.indicatorUnit} variant="body2">{kpiTextConfiguration.unit}</Typography>
      </div>
      <div className={classes.comparisonContainer}>
        <Typography className={classNames(classes.comparisons, classes.spacingToTheRight)}>{kpiTextConfiguration.comparisonIndicator}</Typography>
        <Typography className={classNames(classes.comparisons, classes.infoText)}>{kpiTextConfiguration.comparisonIndicatorInfo}</Typography>
      </div>
    </div>
  );
};

HvKpi.propTypes = {
  /**
   * Class names to be applied.
   */
  className: PropTypes.string,
  /**
   * A Jss Object used to override or extend the styles applied to the button.
   */
  classes: PropTypes.instanceOf(Object).isRequired,
  kpiTextConfiguration: PropTypes.instanceOf(
    PropTypes.shape({
      title: PropTypes.string,
      indicator: PropTypes.string,
      unit: PropTypes.string,
      comparisonIndicatorInfo: PropTypes.string,
      comparisonIndicator: PropTypes.string,
    })
  ).isRequired
};

HvKpi.defaultProps = {
  className: ""
};

export default HvKpi;
