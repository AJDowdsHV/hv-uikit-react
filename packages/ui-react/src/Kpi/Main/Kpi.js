/**
 * Copyright (c) 2018 Hitachi Vantara Corporation.
 *
 *  The copyright to the computer software herein is the property of
 *  Hitachi Vantara Corporation. The software may be used and/or copied only
 *  with the written permission of Hitachi Vantara Corporation or in accordance
 *  with the terms and conditions stipulated in the agreement/contract
 *  under which the software has been supplied.
 */

import React from "react";
import PropTypes from "prop-types";
import classNames from "classnames";
import Typography from "@material-ui/core/Typography";

const HvKpi = props => {
  const {
    classes,
    kpiTextConfiguration,
    visualIndicator,
    visualComparison
  } = props;

  return (
    <div className={classes.kpiContainer}>
      <Typography className={classes.titleText} variant="subtitle2">{kpiTextConfiguration.title}</Typography>
      <div className={classes.indicatorsContainer}>
        { 
          visualIndicator != null &&
          <div className={classNames(classes.visualIndicatorContainer, classes.spacingToTheRight)}>{visualIndicator}</div>
        }
        <Typography className={classNames(classes.indicatorText, classes.spacingToTheRight)} variant="h1">{kpiTextConfiguration.indicator}</Typography>
        <Typography className={classes.indicatorUnit} variant="body2">{kpiTextConfiguration.unit}</Typography>
      </div>
      <div className={classes.comparisonContainer}>
        { 
          visualComparison != null &&
          <div className={classNames(classes.comparisons, classes.visualComparisonContainer, classes.spacingToTheRight)}>{visualComparison}</div>
        }
        <Typography className={classNames(classes.comparisons, classes.infoText)}>{kpiTextConfiguration.comparisonIndicatorInfo}</Typography>
      </div>
    </div>
  );
};

HvKpi.propTypes = {
  /**
   * A Jss Object used to override or extend the styles applied to the button.
   */
  classes: PropTypes.instanceOf(Object).isRequired,
  /**
   * An Element that will be rendered to the left of the kpi indicator text.
   */
  visualIndicator: PropTypes.node,
  /**
   * An Element that will be rendered below the kpi indicator text.
   */
  visualComparison: PropTypes.node,
  /**
   * The object that contains the different labels inside the kpi.
   * 
   * - Title: The text at the top of the kpi.
   * - Indicator: The text in the middle of the kpi.
   * - Unit: The text to the right of the indicator.
   * - comparisonIndicatorInfo: the text to the right of the visual comparison.
   */
  kpiTextConfiguration: PropTypes.shape({
      title: PropTypes.string,
      indicator: PropTypes.string,
      unit: PropTypes.string,
      comparisonIndicatorInfo: PropTypes.string,
    }).isRequired
};

HvKpi.defaultProps = {
  visualIndicator: null,
  visualComparison: null
};

export default HvKpi;
