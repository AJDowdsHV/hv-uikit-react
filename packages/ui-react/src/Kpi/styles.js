/**
 * Copyright (c) 2018 Hitachi Vantara Corporation.
 *
 *  The copyright to the computer software herein is the property of
 *  Hitachi Vantara Corporation. The software may be used and/or copied only
 *  with the written permission of Hitachi Vantara Corporation or in accordance
 *  with the terms and conditions stipulated in the agreement/contract
 *  under which the software has been supplied.
 */

const styles = theme => ({
  kpiContainer:{
    width: "100%",
    height: "100%",
    float: "left"
  },
  visualIndicatorContainer: {
    height: "40px",
    backgroundColor: "transparent",
    display: "inline-flex",
    alignItems: "center",
    justifyContent: "center"
  },
  comparisonContainer: {
    minHeight: "16px",
    display: "flex",
    alignItems: "flex-end",
  },
  visualComparisonContainer: {
    fontFamily: theme.typography.fontFamily,
    ...theme.typography.subtitle1
  },
  indicatorsContainer:{
    display: "inline-flex",
    minHeight: "16px",
    alignItems: "flex-end",
    marginTop: `9px`,
    marginBottom: `1px`
  },
  indicatorUnit: {
    alignSelf: "flex-end",
    paddingBottom: "3px",
    fontFamily: theme.typography.fontFamily,
    ...theme.typography.body2
  },
  spacingToTheRight: {
    marginRight: `${theme.spacing.xs}px`
  },
  comparisons:{
    float: "left"
  },
  indicatorText: {
    fontFamily: theme.typography.fontFamily,
    ...theme.typography.h1
  },
  infoText: {
    fontFamily: theme.typography.fontFamily,
    ...theme.typography.visualization
  },
  titleText: {
    fontFamily: theme.typography.fontFamily,
    ...theme.typography.subtitle2
  }
});

export default styles;
