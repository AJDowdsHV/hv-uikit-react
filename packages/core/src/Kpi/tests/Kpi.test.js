/* eslint-env jest */

import React from "react";
import { mount } from "enzyme";

import HvProvider from "../../Provider";
import Kpi from "..";

const labels = {
  title: "Avg. service time",
  indicator: "8.85",
  unit: "MS",
  comparisonIndicatorInfo: "vs last 24h."
};

describe("Kpi", () => {
  let wrapper;

  beforeEach(async () => {
    wrapper = mount(
      <HvProvider>
        <Kpi labels={labels} />
      </HvProvider>
    );
  });

  it("should be defined", () => {
    expect(wrapper).toBeDefined();
  });

  it("should render correctly", () => {
    expect(wrapper.find(Kpi)).toMatchSnapshot();
  });

  it("should render the Kpi component", () => {
    const KpiComponent = wrapper.find(Kpi);
    expect(KpiComponent.length).toBe(1);
  });
});
