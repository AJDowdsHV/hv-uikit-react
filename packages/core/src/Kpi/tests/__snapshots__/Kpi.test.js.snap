// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Kpi should render correctly 1`] = `
<WithStyles(WithLabels)
  labels={
    Object {
      "comparisonIndicatorInfo": "vs last 24h.",
      "indicator": "8.85",
      "title": "Avg. service time",
      "unit": "MS",
    }
  }
>
  <WithLabels
    classes={
      Object {
        "comparisonComposition": "HvKpi-comparisonComposition",
        "comparisonContainer": "HvKpi-comparisonContainer",
        "indicatorText": "HvKpi-indicatorText",
        "indicatorUnit": "HvKpi-indicatorUnit",
        "indicatorsContainer": "HvKpi-indicatorsContainer",
        "root": "HvKpi-root",
        "spacingToTheRight": "HvKpi-spacingToTheRight",
        "trendLine": "HvKpi-trendLine",
        "visualIndicatorContainer": "HvKpi-visualIndicatorContainer",
      }
    }
    labels={
      Object {
        "comparisonIndicatorInfo": "vs last 24h.",
        "indicator": "8.85",
        "title": "Avg. service time",
        "unit": "MS",
      }
    }
  >
    <HvKpi
      classes={
        Object {
          "comparisonComposition": "HvKpi-comparisonComposition",
          "comparisonContainer": "HvKpi-comparisonContainer",
          "indicatorText": "HvKpi-indicatorText",
          "indicatorUnit": "HvKpi-indicatorUnit",
          "indicatorsContainer": "HvKpi-indicatorsContainer",
          "root": "HvKpi-root",
          "spacingToTheRight": "HvKpi-spacingToTheRight",
          "trendLine": "HvKpi-trendLine",
          "visualIndicatorContainer": "HvKpi-visualIndicatorContainer",
        }
      }
      labels={
        Object {
          "comparisonIndicatorInfo": "vs last 24h.",
          "indicator": "8.85",
          "title": "Avg. service time",
          "unit": "MS",
        }
      }
    >
      <div
        className="HvKpi-root"
      >
        <div>
          <WithStyles(ForwardRef)
            variant="highlightText"
          >
            <ForwardRef
              classes={
                Object {
                  "3xlTitle": "HvTypography-3xlTitle",
                  "4xlTitle": "HvTypography-4xlTitle",
                  "5xlTitle": "HvTypography-5xlTitle",
                  "disabledButtonText": "HvTypography-disabledButtonText",
                  "highlightText": "HvTypography-highlightText",
                  "infoText": "HvTypography-infoText",
                  "inlineLink": "HvTypography-inlineLink",
                  "lTitle": "HvTypography-lTitle",
                  "labelText": "HvTypography-labelText",
                  "mTitle": "HvTypography-mTitle",
                  "normalText": "HvTypography-normalText",
                  "placeholderText": "HvTypography-placeholderText",
                  "root": "HvTypography-root",
                  "sLink": "HvTypography-sLink",
                  "sText": "HvTypography-sText",
                  "sTitle": "HvTypography-sTitle",
                  "selectedNavText": "HvTypography-selectedNavText",
                  "selectedText": "HvTypography-selectedText",
                  "vizText": "HvTypography-vizText",
                  "xlTitle": "HvTypography-xlTitle",
                  "xsTitle": "HvTypography-xsTitle",
                  "xxlTitle": "HvTypography-xxlTitle",
                  "xxsTitle": "HvTypography-xxsTitle",
                }
              }
              variant="highlightText"
            >
              <p
                className="HvTypography-root HvTypography-highlightText"
              >
                Avg. service time
              </p>
            </ForwardRef>
          </WithStyles(ForwardRef)>
        </div>
        <div
          className="HvKpi-indicatorsContainer"
        >
          <WithStyles(ForwardRef)
            className="HvKpi-spacingToTheRight HvKpi-indicatorText"
            variant="lTitle"
          >
            <ForwardRef
              className="HvKpi-spacingToTheRight HvKpi-indicatorText"
              classes={
                Object {
                  "3xlTitle": "HvTypography-3xlTitle",
                  "4xlTitle": "HvTypography-4xlTitle",
                  "5xlTitle": "HvTypography-5xlTitle",
                  "disabledButtonText": "HvTypography-disabledButtonText",
                  "highlightText": "HvTypography-highlightText",
                  "infoText": "HvTypography-infoText",
                  "inlineLink": "HvTypography-inlineLink",
                  "lTitle": "HvTypography-lTitle",
                  "labelText": "HvTypography-labelText",
                  "mTitle": "HvTypography-mTitle",
                  "normalText": "HvTypography-normalText",
                  "placeholderText": "HvTypography-placeholderText",
                  "root": "HvTypography-root",
                  "sLink": "HvTypography-sLink",
                  "sText": "HvTypography-sText",
                  "sTitle": "HvTypography-sTitle",
                  "selectedNavText": "HvTypography-selectedNavText",
                  "selectedText": "HvTypography-selectedText",
                  "vizText": "HvTypography-vizText",
                  "xlTitle": "HvTypography-xlTitle",
                  "xsTitle": "HvTypography-xsTitle",
                  "xxlTitle": "HvTypography-xxlTitle",
                  "xxsTitle": "HvTypography-xxsTitle",
                }
              }
              variant="lTitle"
            >
              <h2
                className="HvKpi-spacingToTheRight HvKpi-indicatorText HvTypography-root HvTypography-lTitle"
              >
                8.85
              </h2>
            </ForwardRef>
          </WithStyles(ForwardRef)>
          <WithStyles(ForwardRef)
            className="HvKpi-indicatorUnit"
            variant="sTitle"
          >
            <ForwardRef
              className="HvKpi-indicatorUnit"
              classes={
                Object {
                  "3xlTitle": "HvTypography-3xlTitle",
                  "4xlTitle": "HvTypography-4xlTitle",
                  "5xlTitle": "HvTypography-5xlTitle",
                  "disabledButtonText": "HvTypography-disabledButtonText",
                  "highlightText": "HvTypography-highlightText",
                  "infoText": "HvTypography-infoText",
                  "inlineLink": "HvTypography-inlineLink",
                  "lTitle": "HvTypography-lTitle",
                  "labelText": "HvTypography-labelText",
                  "mTitle": "HvTypography-mTitle",
                  "normalText": "HvTypography-normalText",
                  "placeholderText": "HvTypography-placeholderText",
                  "root": "HvTypography-root",
                  "sLink": "HvTypography-sLink",
                  "sText": "HvTypography-sText",
                  "sTitle": "HvTypography-sTitle",
                  "selectedNavText": "HvTypography-selectedNavText",
                  "selectedText": "HvTypography-selectedText",
                  "vizText": "HvTypography-vizText",
                  "xlTitle": "HvTypography-xlTitle",
                  "xsTitle": "HvTypography-xsTitle",
                  "xxlTitle": "HvTypography-xxlTitle",
                  "xxsTitle": "HvTypography-xxsTitle",
                }
              }
              variant="sTitle"
            >
              <h4
                className="HvKpi-indicatorUnit HvTypography-root HvTypography-sTitle"
              >
                MS
              </h4>
            </ForwardRef>
          </WithStyles(ForwardRef)>
        </div>
      </div>
    </HvKpi>
  </WithLabels>
</WithStyles(WithLabels)>
`;
